lib/runtime/invalid_transaction.go

```diff
@@ -0,0 +1,146 @@
+)
+
+// InvalidTransaction is child VDT of TransactionValidityError
+type InvalidTransaction scale.VaryingDataType
+
+// Index fulfils the VaryingDataTypeValue interface.  T
+func (InvalidTransaction) Index() uint {
+	return 0
+}
+
+var (
+	errUnexpectedTxCall         = errors.New("call of the transaction is not expected")
+	errInvalidPayment           = errors.New("invalid payment")
+	errInvalidTransaction       = errors.New("invalid transaction")
+	errOutdatedTransaction      = errors.New("outdated transaction")
+	errBadProof                 = errors.New("bad proof")
+	errAncientBirthBlock        = errors.New("ancient birth block")
+	errExhaustsResources        = errors.New("exhausts resources")
+	errMandatoryDispatchError   = errors.New("mandatory dispatch error")
+	errInvalidMandatoryDispatch = errors.New("invalid mandatory dispatch")
+)
+
+// Call The call of the transaction is not expected
+type Call struct{}
+
+// Index Returns VDT index
+func (Call) Index() uint { return 0 }
+
+// Payment General error to do with the inability to pay some fees (e.g. account balance too low)
+type Payment struct{}
+
+// Index Returns VDT index
+func (Payment) Index() uint { return 1 }
+
+// Future General error to do with the transaction not yet being valid (e.g. nonce too high)
+type Future struct{}
+
+// Index Returns VDT index
+func (Future) Index() uint { return 2 }
+
+// Stale General error to do with the transaction being outdated (e.g. nonce too low)
+type Stale struct{}
+
+// Index Returns VDT index
+func (Stale) Index() uint { return 3 }
+
+// BadProof General error to do with the transaction’s proofs (e.g. signature)
+type BadProof struct{}
+
+// Index Returns VDT index
+func (BadProof) Index() uint { return 4 }
+
+// AncientBirthBlock The transaction birth block is ancient
+type AncientBirthBlock struct{}
+
+// Index Returns VDT index
+func (AncientBirthBlock) Index() uint { return 5 }
+
+// ExhaustsResources The transaction would exhaust the resources of current block
+type ExhaustsResources struct{}
+
+// Index Returns VDT index
+func (ExhaustsResources) Index() uint { return 6 }
+
+var invalidCustom InvalidCustom
+
+// InvalidCustom Any other custom invalid validity that is not covered
+type InvalidCustom uint8
+
+// Index Returns VDT index
+func (InvalidCustom) Index() uint { return 7 }
+
+// BadMandatory An extrinsic with a Mandatory dispatch resulted in Error
+type BadMandatory struct{}
+
+// Index Returns VDT index
+func (BadMandatory) Index() uint { return 8 }
+
+// MandatoryDispatch A transaction with a mandatory dispatch
+type MandatoryDispatch struct{}
+
+// Index Returns VDT index
+func (MandatoryDispatch) Index() uint { return 9 }
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
+func (i *InvalidTransaction) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*i)
+	err = vdt.Set(val)
+	if err != nil {
+		return
+	}
+	*i = InvalidTransaction(vdt)
+	return
```

Comment:
nit naked returns are meh (don't tell Tim I said that)

```suggestion
	return nil
```